Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -409,6 +409,25 @@ RefPtr<Gfx::Bitmap> Bitmap::flipped(Gfx::Orientation orientation) const
     return new_bitmap;
 }
 
+RefPtr<Gfx::Bitmap> Bitmap::cropped(Gfx::Point<int> top_left, Gfx::Point<int> bottom_right) const
+{
+    auto crop_width = top_left.absolute_relative_distance_to(bottom_right).x();
+    auto crop_height = top_left.absolute_relative_distance_to(bottom_right).y();
+
+    auto new_bitmap = Gfx::Bitmap::create(this->format(), { crop_width, crop_height }, scale());
+    if (!new_bitmap)
+        return nullptr;
+
+    for (int i = top_left.x(); i < crop_width; i++) {
```

Comment:
Why not `blit()`? That way we can use fast paths or fall back to slower paths (if we ever add e.g. cropping to a different format).

Reply:
Im not at all familiar with the blit function. At the moment i am just providing a basic way of achieving this functionality but over the next few days ill see if I can improve it using blit :). Thanks for the heads up